src/compile/selection/transforms/project.ts

```diff
@@ -152,16 +161,15 @@ const project: TransformCompiler = {
       }
     }
 
-    if (selDef.init) {
+    if (init) {
       const parseInit = <T extends SelectionInit | SelectionInitInterval>(i: Dict<T>): T[] => {
         return proj.items.map(p => (i[p.channel] !== undefined ? i[p.channel] : i[p.field]));
       };
 
-      if (selDef.type === 'interval') {
-        selCmpt.init = parseInit(selDef.init);
+      if (type === 'interval') {
+        selCmpt.init = parseInit(selDef.value as SelectionInitIntervalMapping);
```

Comment:
Why is the type case needed here? Is `selDef.value` not compatible with what `parseInit` expects?

Reply:
The type guard broke because we now read the selection type off `selCmpt` not `selDef`. And I wasn't able to get the type guard working again with alternate forms of the `if` statement. Definitely open to feedback though.

`if (isString(selDef.select) ? selDef.select === 'interval' : isObject(selDef.select) ? selDef.select.type === 'interval' : false) {`

Reply:
I see. It's hard to say for me without diving into the types so let's defer fixing this until we refactor the types of `BaseSelectionDef` as discussed below.

Reply:
That sounds good to me :)